### PR TITLE
configurable temptations

### PR DESCRIPTION
**Pull request by Oharass**
I don't know why this PurPur does not allow everyone and my server and others to customize mob attraction magnet temptation goals for mobs. This adds

CONFIG...
```yaml
ocelot: 
  attributes: 
    max_health: 10.0
  breeding-delay-ticks: 6000
  ridable: false
  ridable-in-water: false
  goal: 
    tempt:
      enabled: true
      items: 
        - "minecraft:carrot"
        - "minecraft:wheat"
      scared: false
      speed: 1.5
```

example

it add all not air if you use "*".

air not permitted to enter list ever

tell me any change. i know it a lot and I am new to PurPur program but learning

### CODE REVIEW

**Comment by Misio12320** [comment]
May a chance to natural attack player 🤔

**Comment by Oharass** [comment]
this may be later. just temptation now this adds

**Comment by Oharass** [comment]
some mob already configured to attack player config. Not all

File: patches/server/0258-configurable-mob-tempt-goals.patch [hunk 1]
```diff
@@ -0,0 +1,2310 @@
...
+@@ -112,4 +112,12 @@ public class GlowSquid extends Squid {
+     public int getDarkTicksRemaining() {
+         return this.entityData.get(DATA_DARK_TICKS_REMAINING);
+     }
++
++    // Purpur start
++    @Override
++    protected void registerGoals() {
++        super.registerGoals();
++        if (this.level.purpurConfig.glowSquidTemptEnabled) this.goalSelector.addGoal(5, new net.minecraft.world.entity.ai.goal.TemptGoal(this, this.level.purpurConfig.glowSquidTemptSpeed, net.minecraft.world.item.crafting.Ingredient.of(this.level.purpurConfig.glowSquidTemptItems.toArray(new net.minecraft.world.item.Item[0])), this.level.purpurConfig.glowSquidTemptScared)); // Purpur
```
Comment (BillyGalbreath) [inline patches/server/0258-configurable-mob-tempt-goals.patch]: we should build and store the `Ingredient` in the config so we're not building it on each entity spawn. entity spawn is already heavy enough on servers

File: patches/server/0258-configurable-mob-tempt-goals.patch [hunk 2]
```diff
@@ -0,0 +1,2310 @@
...
+ }
+diff --git a/src/main/java/net/minecraft/world/entity/animal/Bee.java b/src/main/java/net/minecraft/world/entity/animal/Bee.java
+index d4aa4a33f52b9dbc0756f7a6024105e0c2b9983f..80fd225118146c772a9bcc777a63e0c58b76dd8f 100644
+--- a/src/main/java/net/minecraft/world/entity/animal/Bee.java
++++ b/src/main/java/net/minecraft/world/entity/animal/Bee.java
+@@ -255,6 +255,7 @@ public class Bee extends Animal implements NeutralMob, FlyingAnimal {
+         this.targetSelector.addGoal(1, (new Bee.BeeHurtByOtherGoal(this)).setAlertOthers(new Class[0]));
+         this.targetSelector.addGoal(2, new Bee.BeeBecomeAngryTargetGoal(this));
+         this.targetSelector.addGoal(3, new ResetUniversalAngerTargetGoal<>(this, true));
++        if (this.level.purpurConfig.beeTemptEnabled) this.goalSelector.addGoal(5, new net.minecraft.world.entity.ai.goal.TemptGoal(this, this.level.purpurConfig.beeTemptSpeed, net.minecraft.world.item.crafting.Ingredient.of(this.level.purpurConfig.beeTemptItems.toArray(new net.minecraft.world.item.Item[0])), this.level.purpurConfig.beeTemptScared)); // Purpur
```
Comment (BillyGalbreath) [inline patches/server/0258-configurable-mob-tempt-goals.patch]: you're adding a 2nd `TempGoal` to a lot of these entities. we should be editing the existing tempt goal if already present

File: patches/server/0258-configurable-mob-tempt-goals.patch [hunk 3]
```diff
@@ -0,0 +1,2310 @@
...
+     @Override
+diff --git a/src/main/java/net/minecraft/world/entity/animal/Cat.java b/src/main/java/net/minecraft/world/entity/animal/Cat.java
+index 0d521af1a44acc0fd62f209b845b6d128e42baf4..2e83731c39f1db77e5c48a90dfe0d56f353daf27 100644
+--- a/src/main/java/net/minecraft/world/entity/animal/Cat.java
++++ b/src/main/java/net/minecraft/world/entity/animal/Cat.java
+@@ -174,6 +174,7 @@ public class Cat extends TamableAnimal {
+         this.targetSelector.addGoal(1, new net.pl3x.purpur.entity.ai.HasRider(this)); // Purpur
+         this.targetSelector.addGoal(1, new NonTameRandomTargetGoal<>(this, Rabbit.class, false, (Predicate) null));
+         this.targetSelector.addGoal(1, new NonTameRandomTargetGoal<>(this, Turtle.class, false, Turtle.BABY_ON_LAND_SELECTOR));
++        if (this.level.purpurConfig.catTemptEnabled) this.goalSelector.addGoal(5, new net.minecraft.world.entity.ai.goal.TemptGoal(this, this.level.purpurConfig.catTemptSpeed, net.minecraft.world.item.crafting.Ingredient.of(this.level.purpurConfig.catTemptItems.toArray(new net.minecraft.world.item.Item[0])), this.level.purpurConfig.catTemptScared)); // Purpur
```
Comment (BillyGalbreath) [inline patches/server/0258-configurable-mob-tempt-goals.patch]: looks like you copy pasted the priority `5` in all of these. the placement of this line and the priority number matter a lot. each entity needs to be carefully thought out for this.

**Comment by Oharass** [comment]
I have made requested changes. I agree this makes much better. Temptation priority is now taylor fit per mob. 3 seem good fit for a lot, however many need 1-5 depending on other goal and behavior.

**Comment by Oharass** [comment]
I have decided I do not like this. I may try again in future with different approach trying. Current way is bad idea, too hard to maintain and BIG DIFF. This removes.

**Pull request closed without merging**